stdlib/public/runtime/ProtocolConformance.cpp

```diff
@@ -485,6 +488,18 @@ swift_conformsToProtocolImpl(const Metadata *const type,
         C.cacheResult(matchingType, protocol, witness, /*always cache*/ 0);
       }
     }
+  };
+
+  auto snapshot = C.SectionsToScan.snapshot();
+  if (C.scanSectionsBackwards) {
```

Comment:
Might be nice to do something like:

```
if (C.scanSectionsBackwards)
  std::reverse(std::begin(snapshot), std::end(snapshot))
for (auto &section : snapshot)
  processSection(section);
```

Reply:
As discussed offline, we can't mutate in place, but we settled on:
```
  auto snapshot = C.SectionsToScan.snapshot();
  if (C.scanSectionsBackwards) {
    for (auto &section : llvm::reverse(snapshot))
      processSection(section);
  } else {
    for (auto &section : snapshot)
      processSection(section);
  }
```